python/pyspark/sql/connect/column.py

```diff
@@ -14,20 +14,75 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import types
 import uuid
-from typing import cast, get_args, TYPE_CHECKING, Callable, Any
+from typing import cast, get_args, TYPE_CHECKING, Callable, Any, Union
 
 import decimal
 import datetime
 
 import pyspark.sql.connect.proto as proto
 from pyspark.sql.connect._typing import PrimitiveType
+from pyspark.sql.types import DataType
+import pyspark.sql.types as sqlTypes
+import pyspark.sql.types as sql_types
 
 if TYPE_CHECKING:
     from pyspark.sql.connect.client import RemoteSparkSession
     import pyspark.sql.connect.proto as proto
 
 
+def _string_to_sql_type(dt: str) -> DataType:
```

Comment:
it can not support types like ArrayType, MapType, etc here

if we can continue depending on `pyspark.sql`, what about reusing `_parse_datatype_json_value` or `fromJson` in python/pyspark/sql/types.py?